gpflow/config/__config__.py

```diff
@@ -2,50 +2,106 @@
 import enum
 import os
 from dataclasses import dataclass, field, replace
-from typing import Dict, Optional, Union
+from typing import Dict, Optional
 
 import numpy as np
 import tabulate
 import tensorflow as tf
 import tensorflow_probability as tfp
 
-
 __all__ = [
-    "Config", "as_context", "config", "set_config",
-    "default_float", "set_default_float",
-    "default_int", "set_default_int",
-    "default_jitter", "set_default_jitter",
-    "default_positive_bijector", "set_default_positive_bijector",
-    "default_positive_minimum", "set_default_positive_minimum",
-    "default_summary_fmt", "set_default_summary_fmt",
-    "positive_bijector_type_map"
+    "Config",
+    "as_context",
+    "config",
+    "set_config",
+    "default_float",
+    "set_default_float",
+    "default_int",
+    "set_default_int",
+    "default_jitter",
+    "set_default_jitter",
+    "default_positive_bijector",
+    "set_default_positive_bijector",
+    "default_positive_minimum",
+    "set_default_positive_minimum",
+    "default_summary_fmt",
+    "set_default_summary_fmt",
+    "positive_bijector_type_map",
 ]
 
-
 __config = None
 
 
 class _Values(enum.Enum):
     """Setting's names collection with default values. The `name` method returns name
     of the environment variable. E.g. for `SUMMARY_FMT` field the environment variable
     will be `GPFLOW_SUMMARY_FMT`."""
+
     INT = np.int32
     FLOAT = np.float64
     POSITIVE_BIJECTOR = "softplus"
     POSITIVE_MINIMUM = None
-    SUMMARY_FMT = None
+    SUMMARY_FMT = "fancy_grid"
     JITTER = 1e-6
 
     @property
     def name(self):
         return f"GPFLOW_{super().name}"
 
 
-def default(value: _Values):
+def _default(value: _Values):
     """Checks if value is set in the environment."""
     return os.getenv(value.name, default=value.value)
 
 
+def _default_numeric_type_factory(valid_types, enum_key, type_name):
+    value = _default(enum_key)
+    if value in valid_types.values():
+        return value
+    if value not in valid_types:
+        raise TypeError(f"Config cannot recognize {type_name} type.")
+    return valid_types[value]
+
+
+def _default_int_factory():
+    valid_types = dict(int16=np.int16, int32=np.int32, int64=np.int64)
+    return _default_numeric_type_factory(valid_types, _Values.INT, "int")
+
+
+def _default_float_factory():
+    valid_types = dict(float16=np.float16, float32=np.float32, float64=np.float64)
+    return _default_numeric_type_factory(valid_types, _Values.FLOAT, "float")
+
+
+def _default_jitter_factory():
+    try:
+        value = float(_default(_Values.JITTER))
+    except ValueError:
+        raise TypeError("Config cannot set the jitter value with non float type.")
+    return value
+
+
+def _default_positive_bijector_factory():
+    bijector_type = _default(_Values.POSITIVE_BIJECTOR)
+    if bijector_type not in positive_bijector_type_map().keys():
+        raise TypeError("Config cannot set the passed value as a default positive bijector.")
```

Comment:
might be helpful to include in the exception message what the read-in value was and which keys are valid ?

Reply:
I don't it is necessary to match a full message. Messages are different and comparing all of them adds extra complexity to the testing, _it can be messy_. Message can change and it does not necessarily mean that we need to update tests, it doesn't change the framework's behaviour after all. _The review of the message content is much more important._

Reply:
Not in the test, in *here*!

Reply:
```suggestion
        valid_values = list(positive_bijector_type_map.keys())
        raise TypeError(f"Config cannot set the passed value {bijector_type} as a default positive bijector. Valid values: {valid_values}")
```